src/v/rpc/simple_protocol.cc

```diff
@@ -153,6 +153,7 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
                     ctx->res.conn->addr);
                   rs.probe().method_not_found();
                   netbuf reply_buf;
+                  reply_buf.set_version(ctx->get_header().version);
```

Comment:
This makes me think it'd be nice to be able to add a `vassert` that we never send a v0 `netbuf` over the wire, e.g. in `send_reply()`. I don't think we can though since we'll still send v0 over the wire if talking to an old server (though it gets ignored regardless)

Another thought is that we could use some `simple_reply_buf` around the simple protocol instead of `netbuf` that is constructed with an initial version (even if it's just `_version`).

I don't feel strongly that either of these would be much better than what's there, but just brainstorming ways we can be sure our sent bytes are reasonable

Reply:
> I don't think we can though since we'll still send v0 over the wire if talking to an old server (though it gets ignored regardless)

correct

> Another thought is that we could use some simple_reply_buf around the simple protocol instead of netbuf that is constructed with an initial version (even if it's just _version).

i think the right move would be to add a constructor to netbuf so it can't be constructed without providing a version

Reply:
Yea that seems reasonable too. I was initially worried we'd end up with some lifecycle issues, e.g. not knowing what the right version is at time of construction, but I agree baking this into netbuf would feel better